src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -41,16 +59,56 @@ internal class MessageReceiverExample(private val callback: (message: String, md
     }
 
     fun processMessage(message: PubsubMessage) {
-        val messageContent = message.data.toStringUtf8()
-        val md5sum = this.dockerProcessor.getMd5Sum(messageContent)
+        val request = CompilerFleet.CompilerFleetRequest.parseFrom(message.data)
+
+        val destination = this.tasksDir.resolve(request.taskId).resolve("files")
+        val zipStream = ZipInputStream(ByteArrayInputStream(request.zipBytes.toByteArray()))
+        var entry: ZipEntry? = zipStream.nextEntry
+
+        while (entry != null) {
+            val filename = entry.name
+            val newFile = destination.resolve(filename).toFile()
+
+            if (!newFile.parentFile.mkdirs()) {
+                throw IOException("unable to create dirs while unzipping")
+            }
+
+            FileOutputStream(newFile).use {
+                ByteStreams.copy(zipStream, it)
+            }
+
+            entry = zipStream.nextEntry
+        }
+
+        val rootFileName = request.rootFileName
+        val rootFile = destination.resolve(rootFileName).toFile()
+        if (!rootFile.exists()) {
+            throw IOException("path to root doesn't exists")
+        }
 
-        this.callback(messageContent, md5sum)
+        this.callback("md5 sum of root file", dockerProcessor.getMd5Sum(rootFile))
     }
 }
 
-class SubscribeManager(subscriptionId: String, callback: (message: String, md5sum: String) -> Unit) {
+class SubscribeManager(tasksDir: String,
+                       subscriptionId: String,
+                       callback: (message: String, md5sum: String) -> Unit) {
     private val subscriptionName = SubscriptionName.of(PROJECT_ID, subscriptionId)
-    private val receiver = MessageReceiverExample(callback)
+    private val receiver: TaskReceiver
+
+    init {
+        val directory = Paths.get(tasksDir)
+
+        if (!directory.toFile().exists()) {
+            throw IOException("tasksDir directory doesn't exists")
```

Comment:
Write the value from the argument. If I made a typo in my shell scripts which launch this code, and the value of tasksDir is "" then printing will immediately give me a clue.

Reply:
Added name to print.
Done